### PR TITLE
[MIG][11.0] point_of_sale: migration scripts

### PR DESCRIPTION
**Pull request by cubells**
Depends on:

- [x] stock

cc @Tecnativa

### CODE REVIEW

File: addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt
Comment (pedrobaeza) [inline addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt:23]: You have to rename column on pre for not losing values (they are incompatible types).

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt
Comment (cubells) [reply addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt:23]: @pedrobaeza 

done!

File: addons/point_of_sale/migrations/11.0.1.0.1/post-migration.py
Comment (pedrobaeza) [inline addons/point_of_sale/migrations/11.0.1.0.1/post-migration.py]: You can put these triple quote on previous line

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Flake8 errors:
```
./addons/point_of_sale/migrations/11.0.1.0.1/post-migration.py:14:37: W291 trailing whitespace
./addons/point_of_sale/migrations/11.0.1.0.1/post-migration.py:16:32: W291 trailing whitespace
```

**Comment by cubells** [comment]
@pedrobaeza 

done!

**Review by pedrobaeza — APPROVED** [review]

File: odoo/openupgrade/doc/source/modules100-110.rst
Comment (StefanRijnhart) [inline odoo/openupgrade/doc/source/modules100-110.rst:386]: ~~blocked by dependency~~ I see the merge is blocked with a tag

**Comment by StefanRijnhart** [comment]
Ah, I see you blocked this with a tag.

File: odoo/openupgrade/doc/source/modules100-110.rst
Comment (pedrobaeza) [reply odoo/openupgrade/doc/source/modules100-110.rst:386]: Yes, I'm not going to merge until the dependency is merged and this rebased. I use the tag for knowing that

File: addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt
Comment (StefanRijnhart) [inline addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt:67]: Please set rescue = true where name contains 'RESCUE FOR'

Comment (cubells) [reply addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt:67]: done @StefanRijnhart !

Comment (StefanRijnhart) [reply addons/point_of_sale/migrations/11.0.1.0.1/openupgrade_analysis_work.txt:67]: Thanks!

**Review by StefanRijnhart — APPROVED** [review]

**Comment by cvinh** [comment]
I tested, I have an error here : 
`bad query: b'ALTER TABLE "point_of_sale" RENAME "iface_tax_included" TO "openupgrade_legacy_11_0_iface_tax_included"'`
addons/account/migrations/11.0.1.1/pre-migration.py should be : 
`openupgrade.rename_columns(
        env.cr, {'pos_config': [('iface_tax_included', None)]})`

Then another error : 
`ValueError: External ID not found in the system: point_of_sale.customer_facing_display_html
load could not load template
Template: point_of_sale.customer_facing_display_html
`

File: addons/point_of_sale/migrations/11.0.1.0.1/pre-migration.py
Comment (pedrobaeza) [inline addons/point_of_sale/migrations/11.0.1.0.1/pre-migration.py]: s/point_of_sale/pos_config

**Comment by pedrobaeza** [comment]
Rebased now that dependencies are merged.

**Comment by cubells** [comment]
@pedrobaeza done!

**Comment by pedrobaeza** [comment]
I already did it... (that's why I put `Rebased` and not `Rebase`), but no special problem.

**Comment by pedrobaeza** [comment]
There's still an error:
```
ValueError: External ID not found in the system: point_of_sale.customer_facing_display_html
```
Please check it

**Comment by ghost** [comment]
Hi, I also have this error: ValueError: External ID not found in the system: point_of_sale.customer_facing_display_html during the migration, has it been fixed?

**Comment by pedrobaeza** [comment]
@cubells please check this or comment me if there's any problem.

**Comment by cubells** [comment]
@pedrobaeza 
I guess the problem here is that the field customer_facing_display_html (https://github.com/OCA/OpenUpgrade/blob/11.0/addons/point_of_sale/models/pos_config.py#L147) is a computed field that tries to render a template (https://github.com/OCA/OpenUpgrade/blob/11.0/addons/point_of_sale/views/point_of_sale.xml#L44) that is not loaded when creating that field.

So this problem raise an error here https://travis-ci.org/OCA/OpenUpgrade/builds/395363373#L4046

Is it possible?

**Comment by cubells** [comment]
@pedrobaeza all is :green_apple:  finally.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**
